auth/src/api/roles.js

```diff
@@ -7,40 +7,78 @@ const handleAsyncErrors = require("../utils/errorHandler");
 const { verifyUser } = require("./../utils/userVerification");
 const { googleAuth } = require("./../utils/getConfigFile");
 
+const getUser = async (req, res) => {
+  if (!req.headers.token) {
+    return sendResponse(res, 400, "Invalid Token");
+  }
+
+  let user = null;
+  const useGoogle = await googleAuth();
+  if (!useGoogle || !req.headers.google || !JSON.parse(req.headers.google)) {
+    // If it is not a google user, it verifies the token is valid and the user exists.
+    user = await verifyUser(req.headers.token);
+    if (user.errorMessage != null) {
+      return sendResponse(res, 400, user.errorMessage);
+    }
+  } else {
+    // If it is a google user, it makes a request to the google API using the token and fetches the email
+    const tokenInfoRes = await fetch(
+      `https://www.googleapis.com/oauth2/v3/tokeninfo?id_token=${req.headers.token}`
+    );
+    const payload = await tokenInfoRes.json();
+    // If the user is in the database it finds it or returns an error message.
+    user = await User.findOne({ email: payload.email, googleAuth: true });
+    if (!user) {
```

Comment:
Maybe this should be outside the else statement so that way we can double check that we aren't returning `null` (might be from the if statement still,. if `verifyUser` fails or smth idk